### PR TITLE
[WIP] Adds ants and ant accessories to the station

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Introduces a new reagent, mob, and all-around general nuisance, the ant.
This started off as a port of Goon's ants, but ended up taking on a mind of its own.
Leaving food out on the floor will attract ants after 5 minutes, and after 10 minutes, that food will crumble into a moldy slurry.*
Ants can also be found very rarely in maint, the bathrooms, and even the kitchen.
What can you do with ants? I'm glad you asked!
You can cover yourself in them!
Inject others with them!
Do `insert activity a typical chemist main would do here` with them!
The possibilities are limited only by your supply of ants.

*Food left on tables, inside storage containers, and in crates/lockers will remain unaffected by ants. This is because, as everyone is well aware, ants are incapable of climbing whatsoever.

## Why It's Good For The Game
https://youtu.be/RpdscsPV1yE

Jokes aside, it adds a little more realism to leaving food just carelessly lying around somewhere, as well as add some neat things to the station in the process.

## Changelog
:cl: Wallem
add: Due to a leak in Nanotrasen Brand Ant Farms during shipping, a large influx of space-ants have been inadvertently dumped onto every Nanotrasen station. Don't go leaving food around, and please be sure to watch your step, we're pretty sure they bite...
imageadd: New giant ant sprite by Doshmobile

/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
Drafting because its 4am and I still gotta iron out some bugs.

**Comment by Mothblocks** [comment]
heh. bugs. ants.

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -967,3 +967,85 @@
...
+		if(DT_PROB(3, delta_time)) //See? It's not ALL bad!
+			M.reagents.add_reagent(/datum/reagent/toxin/histamine,rand(1,3))
+			ants_remaining -= 5
+	ants_remaining -= 1
+	if(ants_remaining <= 0)
+		M.remove_status_effect(STATUS_EFFECT_ANTS) //If this person has no more ants on them, they are no longer affected.
+
+/atom/movable/screen/alert/status_effect/ants
+	name = "Ants!"
+	desc = "<span class='warning'>JESUS FUCKING CHRIST! CLICK TO GET THOSE THINGS OFF!</span>"
```
Comment (Fikou) [inline code/datums/status_effects/debuffs.dm:1039]:
```suggestion
	desc = "<span class='warning'>SPACE JESUS FUCKING CHRIST! CLICK TO GET THOSE THINGS OFF!</span>"
```

**Comment by ArcaneMusic** [comment]
(I really dig this PR)
One more thing: If the ants are listening for a dropped signal to begin decomposition or ant-harvesting or whatever, see if you can spread that behavior to botany crops being harvested. Botany casually having 2000 apples on the floor and having them slowly getting eaten away by hordes of ants is peak realism.

**Comment by Mothblocks** [comment]
as long as this is a component, thats going to take up way too much memory for such a niche feature

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -2611,3 +2611,45 @@
...
+	taste_mult = 1.3
+	taste_description = "<span class='warning'>ANTS OH GOD</span>"
+	metabolization_rate = 5 * REAGENTS_METABOLISM //1u per second
+	glass_name = "glass of ants"
+	glass_desc = "Bottoms up...?"
+	var/ant_damage = 0
+	var/amount_left = 0
+
+/datum/reagent/ants/on_mob_life(mob/living/carbon/M)
+	M.adjustBruteLoss(max(0.5, round((ant_damage * 0.1),0.1)))
```
Comment (ArcaneMusic) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2629]: Assuming a full 50 units of ants are injected or fed to a target, they'll take 103.75 brute damage in 50 seconds. You might to tune this down a bit.

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (Sparkezel) [reply code/datums/status_effects/debuffs.dm:1039]: Better yet, make it read your chosen chaplain deity, narsie if you are cultist and heretic gods if you are a heretic

File: code/datums/status_effects/debuffs.dm [hunk 3]
```diff
@@ -967,3 +967,85 @@
...
+	to_chat(owner, "<span class='notice'>All of the ants are off of your body!</span>")
+	UnregisterSignal(owner, COMSIG_COMPONENT_CLEAN_ACT, .proc/ants_washed)
+	. = ..()
+
+/datum/status_effect/ants/proc/ants_washed()
+	owner.remove_status_effect(STATUS_EFFECT_ANTS)
+	return COMPONENT_CLEANED
+
+/datum/status_effect/ants/tick(delta_time)
+	var/mob/living/carbon/human/M = owner
```
Comment (tf-4) [inline code/datums/status_effects/debuffs.dm:1004]: https://github.com/tgstation/tgstation/pull/56872

File: code/datums/status_effects/debuffs.dm [hunk 4]
```diff
@@ -967,3 +967,85 @@
...
+	if(ants_remaining <= 0)
+		M.remove_status_effect(STATUS_EFFECT_ANTS) //If this person has no more ants on them, they are no longer affected.
+
+/atom/movable/screen/alert/status_effect/ants
+	name = "Ants!"
+	desc = "<span class='warning'>JESUS FUCKING CHRIST! CLICK TO GET THOSE THINGS OFF!</span>"
+	icon_state = "antalert"
+
+/atom/movable/screen/alert/status_effect/ants/Click()
+	var/mob/living/L = owner
```
Comment (tf-4) [inline code/datums/status_effects/debuffs.dm:1043]: https://github.com/tgstation/tgstation/pull/56872

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 5]
```diff
@@ -257,3 +257,29 @@
...
+/obj/effect/decal/cleanable/ants/Initialize(mapload)
+	. = ..()
+	var/scale = (rand(6, 8) / 10) + (rand(2, 5) / 50)
+	transform = matrix(transform, scale, scale, MATRIX_SCALE)
+	setDir(pick(GLOB.cardinals))
+	reagents.add_reagent(/datum/reagent/ants, rand(3, 6))
+	pixel_x = rand(-5, 5)
+	pixel_y = rand(-5, 5)
+
+/obj/effect/decal/cleanable/ants/Crossed(atom/movable/AM)
```
Comment (tf-4) [inline code/game/objects/effects/decals/cleanable/misc.dm:278]: https://github.com/tgstation/tgstation/pull/56872

File: code/modules/mob/living/simple_animal/friendly/ant.dm [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	AddElement(/datum/element/pet_bonus, "buzzes!")
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/mob/living/simple_animal/ant/add_cell_sample()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_COCKROACH, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 7) //They're both insects, might as well make it do SOMETHING
+
+/mob/living/simple_animal/ant/Life(delta_time = SSMOBS_DT, times_fired) //Payback time bitch
+	if((src.loc) && isturf(src.loc))
+		if(!stat && !resting && !buckled)
+			for(var/mob/living/simple_animal/hostile/cockroach/CR in view(1,src))
```
Comment (tf-4) [inline code/modules/mob/living/simple_animal/friendly/ant.dm:50]: https://github.com/tgstation/tgstation/pull/56872

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 7]
```diff
@@ -45,6 +49,43 @@
...
 	AddElement(/datum/element/swabable, CELL_LINE_TABLE_COCKROACH, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 7)
 
+/mob/living/simple_animal/hostile/cockroach/proc/splat() //Forces a cockroach death, used for retribution
+	src.health = 0
+	death()
+
+/mob/living/simple_animal/hostile/cockroach/Life(delta_time = SSMOBS_DT, times_fired) //Natural predators
+	if((src.loc) && isturf(src.loc))
+		if(!stat && !resting && !buckled)
+			for(var/obj/effect/decal/cleanable/ants/A in view(1,src))//Tiny ants are defenseless to this mighty foe
```
Comment (tf-4) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:59]: https://github.com/tgstation/tgstation/pull/56872

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 8]
```diff
@@ -2611,3 +2611,45 @@
...
+	color = "#993333"
+	taste_mult = 1.3
+	taste_description = "<span class='warning'>ANTS OH GOD</span>"
+	metabolization_rate = 5 * REAGENTS_METABOLISM //1u per second
+	glass_name = "glass of ants"
+	glass_desc = "Bottoms up...?"
+	var/ant_damage = 0
+	var/amount_left = 0
+
+/datum/reagent/ants/on_mob_life(mob/living/carbon/M)
```
Comment (tf-4) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2628]: https://github.com/tgstation/tgstation/pull/56872

File: code/datums/status_effects/debuffs.dm (hunk 3 above)
Comment (maxymax13) [reply code/datums/status_effects/debuffs.dm:1004]: sussy variable

File: code/datums/status_effects/debuffs.dm [hunk 9]
```diff
@@ -967,3 +967,85 @@
...
+				if(1)
+					M.say("GET THEM OFF ME!!", forced = /datum/status_effect/ants)
+				if(2)
+					M.say("OH GOD THE ANTS!!", forced = /datum/status_effect/ants)
+		if(DT_PROB(30,delta_time))
+			M.emote("scream")
+		if(DT_PROB(80, delta_time))
+			switch (rand(1, 50))
+				if (1 to 8) //16% Chance (if I made it 15% the other two probabilities wouldn't be integers)
+					var/obj/item/bodypart/head/hed = M.get_bodypart(BODY_ZONE_HEAD)
```
Comment (maxymax13) [inline code/datums/status_effects/debuffs.dm:1019]: lmao why

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 2 above)
Comment (maxymax13) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:2629]: 😋

**Comment by EricZilla** [comment]
Space ants are fun, someones gonna have to update the insect phobia at some point

File: code/datums/components/food/decomposition.dm [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+	return ..()
+
+/datum/component/decomposition/proc/storage_check()
+	SIGNAL_HANDLER
+	is_stored = TRUE
+	STOP_PROCESSING(SSobj, src)
+
+/datum/component/decomposition/proc/table_check(obj/item/food/decomp)
+	SIGNAL_HANDLER
+	if(locate(/obj/structure/table) in decomp.loc || is_stored || locate(/obj/machinery/griddle) in decomp.loc)
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:37]: Tables and griddles preserving food should be a trait they have, not something that is hardcoded into decomposition.

File: code/datums/components/food/decomposition.dm [hunk 11]
```diff
@@ -0,0 +1,64 @@
+//"Don't leave food on the floor, that's how we get ants"
+/datum/component/decomposition
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///How decomposed a specific food item is. This will go rather high due to the 2 second process timer.
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:4]: The timer of SSobj should not affect this at all if your code is proper.

File: code/datums/components/food/decomposition.dm [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+	var/decomposition_level = 0
+	var/is_stored = FALSE
+
+/datum/component/decomposition/Initialize()
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/table_check)
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:11]: Move this to `RegisterWithParent`.

File: code/datums/components/food/decomposition.dm [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+	RegisterSignal(parent, COMSIG_ITEM_PICKUP, .proc/picked_up)
+	RegisterSignal(parent, COMSIG_STORAGE_ENTERED, .proc/storage_check) //Checks if you put it in storage
+	RegisterSignal(parent, COMSIG_CLOSET_CONTENTS_EXPOSED, .proc/table_check)
+	RegisterSignal(parent, COMSIG_TRY_STORAGE_HIDE_ALL, .proc/picked_up) //Checks if it's in a closed closet
+	RegisterSignal(parent, COMSIG_STORAGE_EXITED, .proc/table_check) //Checks if a storage object has been dumped
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+
+/datum/component/decomposition/Destroy()
+	UnregisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/table_check)
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:20]: Move all of this (except the `STOP_PROCESSING`) to `UnregisterFromParent`. Then, make this a list in the form of:

```dm
UnregisterSignal(parent, list(
    COMSIG_ITEM_DROPPED,
    COMSIG_ITEM_PICKUP,
    // etc
))
```

File: code/datums/components/food/decomposition.dm [hunk 14]
```diff
@@ -0,0 +1,64 @@
+//"Don't leave food on the floor, that's how we get ants"
+/datum/component/decomposition
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///How decomposed a specific food item is. This will go rather high due to the 2 second process timer.
+	var/decomposition_level = 0
+	var/is_stored = FALSE
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:6]: Can you document this? It's not completely clear to me while reading.

File: code/datums/components/food/decomposition.dm [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+		is_stored = FALSE //Made false here so storage dumping will still be affected.
+		return
+	START_PROCESSING(SSobj, src)
+
+/datum/component/decomposition/proc/picked_up()
+	SIGNAL_HANDLER
+	STOP_PROCESSING(SSobj, src)
+
+/datum/component/decomposition/process(delta_time, obj/item/food/decomp)
+	decomposition_level += 1 //Gonna fire every 2 seconds, so to find specific values in minutes use (minutes*60)/2.
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:47]: **No.** Use delta_time. Your code should work no matter if SSobj's process time is 1 second or 1 minute.

File: code/datums/components/food/decomposition.dm [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+	if(decomposition_level == 300) //10 minutes
+		new /obj/item/food/badrecipe/moldy(get_turf(parent))
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:49]:
```suggestion
		new /obj/item/food/badrecipe/moldy(parent.loc)
```

File: code/datums/components/food/decomposition.dm [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+		qdel(parent)
+		return
+	if(decomposition_level == 150) //5 minutes
+		new /obj/effect/decal/cleanable/ants(get_turf(parent))
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:53]:
```suggestion
		new /obj/effect/decal/cleanable/ants(parent.loc)
```

File: code/datums/components/food/decomposition.dm [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+		return
+	START_PROCESSING(SSobj, src)
+
+/datum/component/decomposition/proc/picked_up()
+	SIGNAL_HANDLER
+	STOP_PROCESSING(SSobj, src)
+
+/datum/component/decomposition/process(delta_time, obj/item/food/decomp)
+	decomposition_level += 1 //Gonna fire every 2 seconds, so to find specific values in minutes use (minutes*60)/2.
+	if(decomposition_level == 300) //10 minutes
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:48]: Then, change all of these to be proper.

File: code/datums/components/food/decomposition.dm [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+/datum/component/decomposition/proc/examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	if (decomposition_level < 50)
+		return
+	if(decomposition_level >= 50 && decomposition_level < 150)
+		examine_list += "[parent] looks kinda stale."
+	if(decomposition_level >= 150 && decomposition_level < 250)
+		examine_list += "[parent] is starting to look pretty gross."
+	if(decomposition_level >= 250)
+		examine_list += "[parent] looks barely edible."
```
Comment (Mothblocks) [inline code/datums/components/food/decomposition.dm:64]: Make this entire thing something in the form of:

```dm
switch (decomposition_level)
    if (50 to 150)
        // code
    if (150 to 250)
        // etc
```

File: code/datums/status_effects/debuffs.dm [hunk 20]
```diff
@@ -967,3 +967,85 @@
...
+		ants_remaining += amount_left
+	. = ..()
+
+/datum/status_effect/ants/on_remove()
+	ants_remaining = 0
+	to_chat(owner, "<span class='notice'>All of the ants are off of your body!</span>")
+	UnregisterSignal(owner, COMSIG_COMPONENT_CLEAN_ACT, .proc/ants_washed)
+	. = ..()
+
+/datum/status_effect/ants/proc/ants_washed()
```
Comment (Mothblocks) [inline code/datums/status_effects/debuffs.dm:999]: Needs `SIGNAL_HANDLER`.

File: code/datums/status_effects/debuffs.dm [hunk 21]
```diff
@@ -967,3 +967,85 @@
...
+	var/mob/living/carbon/human/M = owner
+	M.adjustBruteLoss(max(0.2, round((ants_remaining * 0.06),0.1))) //More ants means more damage
+	if(!M.stat) //Makes sure people don't scratch at themselves while they're dead
+		if(DT_PROB(15, delta_time))
+			var/ant_scream = pick(1,2)
+			switch(ant_scream)
+				if(1)
+					M.say("GET THEM OFF ME!!", forced = /datum/status_effect/ants)
+				if(2)
+					M.say("OH GOD THE ANTS!!", forced = /datum/status_effect/ants)
```
Comment (Mothblocks) [inline code/datums/status_effects/debuffs.dm:1013]: Change this to just `M.say(pick("message 1",  "message2"), forced = /datum/status_effect/ants)`

File: code/datums/status_effects/debuffs.dm [hunk 22]
```diff
@@ -967,3 +967,85 @@
...
+/atom/movable/screen/alert/status_effect/ants/Click()
+	var/mob/living/L = owner
+	if(!istype(L) || !L.can_resist() || L != owner)
+		return
+	to_chat(L, "<span class='notice'>You start to shake the ants off!</span>")
+	if(!do_after(L, 2 SECONDS, TRUE))
+		return
+	else for (var/datum/status_effect/ants/ant_covered in L.status_effects)
+		to_chat(L, "<span class='notice'>You manage to get some of the ants off!</span>")
+		ant_covered.ants_remaining -=7
```
Comment (Mothblocks) [inline code/datums/status_effects/debuffs.dm:1051]:
```suggestion
		ant_covered.ants_remaining -= 7
```

File: code/datums/status_effects/debuffs.dm [hunk 23]
```diff
@@ -967,3 +967,85 @@
...
+	icon_state = "antalert"
+
+/atom/movable/screen/alert/status_effect/ants/Click()
+	var/mob/living/L = owner
+	if(!istype(L) || !L.can_resist() || L != owner)
+		return
+	to_chat(L, "<span class='notice'>You start to shake the ants off!</span>")
+	if(!do_after(L, 2 SECONDS, TRUE))
+		return
+	else for (var/datum/status_effect/ants/ant_covered in L.status_effects)
```
Comment (Mothblocks) [inline code/datums/status_effects/debuffs.dm:1049]: Remove this else completely, the `return` will prevent it from running.

File: code/datums/status_effects/status_effect.dm [hunk 24]
```diff
@@ -55,7 +55,7 @@
 		owner = null
 	return ..()
 
-/datum/status_effect/process()
+/datum/status_effect/process(delta_time)
```
Comment (Mothblocks) [inline code/datums/status_effects/status_effect.dm:58]: You're not *using* this--you need to pass `delta_time` to `tick`.

File: code/game/objects/effects/decals/cleanable/misc.dm (hunk 5 above)
Comment (Mothblocks) [inline code/game/objects/effects/decals/cleanable/misc.dm:278]: `Crossed` does not exist in our codebase, look at how other consumers do it with `connect_loc`.

Note that there is currently a bug where they'll "cross" with themselves, so you need to check for that.

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 25]
```diff
@@ -257,3 +257,29 @@
...
+	if(isliving(AM))
+		var/mob/living/L = AM
```
Comment (Mothblocks) [inline code/game/objects/effects/decals/cleanable/misc.dm:280]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 26]
```diff
@@ -257,3 +257,29 @@
...
+		if(!(HAS_TRAIT(L, TRAIT_PASSTABLE) || L.buckled))
```
Comment (Mothblocks) [inline code/game/objects/effects/decals/cleanable/misc.dm:281]:
```suggestion
		if(!HAS_TRAIT(L, TRAIT_PASSTABLE) && !L.buckled)
```

File: code/game/objects/effects/spawners/decals.dm [hunk 27]
```diff
@@ -11,5 +11,6 @@
 				/obj/effect/decal/cleanable/wrapping = 5,
 				/obj/effect/decal/cleanable/plastic = 5,
 				/obj/effect/decal/cleanable/glass = 5,
-				/obj/effect/decal/cleanable/dirt = 30
+				/obj/effect/decal/cleanable/dirt = 30,
+				/obj/effect/decal/cleanable/ants = 5
```
Comment (Mothblocks) [inline code/game/objects/effects/spawners/decals.dm:15]:
```suggestion
				/obj/effect/decal/cleanable/ants = 5,
```

Will prevent the blame from getting messed up next time.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 28]
```diff
@@ -144,6 +144,9 @@
 	var/atom/L = drop_location()
 	for(var/atom/movable/AM in src)
 		AM.forceMove(L)
+		for(var/i in src)
+			var/atom/movable/thing = i
```
Comment (Mothblocks) [inline code/game/objects/structures/crates_lockers/closets.dm:148]:
```suggestion
		for(var/atom/movable/thing as anything in src)
```

File: code/__DEFINES/dcs/signals.dm [hunk 29]
```diff
@@ -1251,3 +1251,7 @@
 // Exosca signals
 /// Sent on exoscan failure/manual interruption: ()
 #define COMSIG_EXOSCAN_INTERRUPTED "exoscan_interrupted"
+
+// Closet/crate items ejected signal
+///from closet open()
+#define COMSIG_CLOSET_CONTENTS_EXPOSED "closet_contents_exposed"
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals.dm:1257]: This isn't obvious what this signal actually fires on. I didn't notice until later on that this is called on the *contents*, not the locker.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 30]
```diff
@@ -144,6 +144,9 @@
...
+			SEND_SIGNAL(thing, COMSIG_CLOSET_CONTENTS_EXPOSED, src)
```
Comment (Mothblocks) [inline code/game/objects/structures/crates_lockers/closets.dm:149]: I don't like that we're calling this on every single item inside the closet. I think this should fire on the closet itself, `decomposition` should use `connect_loc`, then if the location is a locker, it registers to it.

File: code/modules/mob/living/simple_animal/friendly/ant.dm [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+	health = 75
+	maxHealth = 75
+	var/mob/living/simple_animal/hostile/cockroach/movement_target
+	//Limits how often the ant can kill cockroaches
+	COOLDOWN_DECLARE(emote_cooldown)
+	var/turns_since_scan = 0
+
+/mob/living/simple_animal/ant/Initialize()
+	. = ..()
+	AddElement(/datum/element/pet_bonus, "buzzes!")
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/friendly/ant.dm:41]: yuck

File: code/modules/mob/living/simple_animal/friendly/ant.dm [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/mob/living/simple_animal/ant/add_cell_sample()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_COCKROACH, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 7) //They're both insects, might as well make it do SOMETHING
+
+/mob/living/simple_animal/ant/Life(delta_time = SSMOBS_DT, times_fired) //Payback time bitch
+	if((src.loc) && isturf(src.loc))
+		if(!stat && !resting && !buckled)
+			for(var/mob/living/simple_animal/hostile/cockroach/CR in view(1,src))
+				if(!CR.stat && Adjacent(CR) && COOLDOWN_FINISHED(src, emote_cooldown))
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/friendly/ant.dm:51]: Early continue.

File: code/modules/mob/living/simple_animal/friendly/ant.dm [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+					manual_emote("chomps \the [CR]!")
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/friendly/ant.dm:52]: `\the` is implicit.

File: code/modules/mob/living/simple_animal/friendly/ant.dm [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+					CR.splat()
+					movement_target = null
+					stop_automated_movement = 0
+					COOLDOWN_START(src, emote_cooldown, 1 MINUTES)
+					break
+
+	..()
+	if(!stat && !buckled)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/friendly/ant.dm:60]: Early return.

File: code/modules/mob/living/simple_animal/friendly/ant.dm [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+		turns_since_scan++
+		if(turns_since_scan > 5)
+			walk_to(src,0)
+			turns_since_scan = 0
+			if((movement_target) && !(isturf(movement_target.loc) || ishuman(movement_target.loc) ))
+				movement_target = null
+				stop_automated_movement = 0
+			if( !movement_target || !(movement_target.loc in oview(src, 3)) )
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/friendly/ant.dm:68]:
```suggestion
			if(!movement_target || !(movement_target.loc in oview(src, 3)))
```

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 36]
```diff
@@ -45,6 +49,43 @@
 /mob/living/simple_animal/hostile/cockroach/add_cell_sample()
 	AddElement(/datum/element/swabable, CELL_LINE_TABLE_COCKROACH, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 7)
 
+/mob/living/simple_animal/hostile/cockroach/proc/splat() //Forces a cockroach death, used for retribution
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:52]: This function looks pointless, callnig `death()` looks equivalent. What is it for?

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 37]
```diff
@@ -45,6 +49,43 @@
...
+	src.health = 0
+	death()
+
+/mob/living/simple_animal/hostile/cockroach/Life(delta_time = SSMOBS_DT, times_fired) //Natural predators
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:56]: This is a copy-paste, this should be its own component/element or, better yet, an AI datum.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 38]
```diff
@@ -2611,3 +2611,45 @@
...
+	M.adjustBruteLoss(max(0.5, round((ant_damage * 0.1),0.1)))
+	if(prob(5)) //Due to the fact this has a chance of happening every cycle, it's more likely to happen than it looks
+		if(prob(1)) //Super rare statement
+			M.say("AUGH NO NOT THE ANTS! NOT THE ANTS! AAAAUUGH THEY'RE IN MY EYES! MY EYES! AUUGH!!", forced = /datum/reagent/ants)
+		else
+			var/scream_statement = pick(1,2)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2634]: Do `say(pick(...))`.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Wallemations** [comment]
I put in a good portion of the suggestions given, and resolved them if they were added.
The other ones, I will admit, go a little over my head.
I'm going to work on figuring those out though!

**Comment by Wallemations** [comment]
Wow nevermind, I just managed to fuck it all up again, give me a second

**Comment by Wallemations** [comment]
So, it turns out my house is being fucking fumigated.
For ants.
Go figure.
So I need to get out in the next 45 minutes and won't be able to actually work on this shit for a week due to a lack of functioning computer in that timeframe.
Good thing I made this a draft.
I can still at least edit directly to the git, but the problem with that is lack of ability to _**test**_ that code myself.
So for now I'm going to close this PR, and I'll re-open it in about a week or so when I get back to this fully functioning computer.

**Comment by maxymax13** [comment]
you made the ants so realistic they became real

**Comment by Wallemations** [comment]
> you made the ants so realistic they became real

And here I thought I was a novice coder

**Comment by GunHog** [comment]
I guess we know now why you made an ant PR...It seemed really inspired for some reason...

**Comment by Mothblocks** [comment]
false bugs, youll soon be alone, see how they drop like a stone, you're free now, forever to dance, in a spiral of ants

**Comment by ArcaneMusic** [comment]
Look I'm going to be real, you really got us in the first half. Hope your bugs issues and your bugs issues get worked out!

**Comment by Twaticus** [comment]
if you get ants in game you get ants in irl

**Comment by Thunder12345** [comment]
Ant PR delayed by actual ants, amazing

**Comment by Wallemations** [comment]
Finally able to get back to working on this.
To do list:

- [ ] Figure out exactly how `connect_loc` functions so I can use it to make lockers/grittles/other food-based machines pause the decomposition timer
- [ ] Mothblocks said food preservation should be a trait tables/griddles/etc have instead of being hard-coded into decomposition. I'm an idiot so I'll have to figure out what that means. If you meant structures should have a variable and instead of searching for that specific structure I should look to see if whatever structure beneath the food has said variable, that's actually much smarter than what I'm doing at the moment and I should probably do that. It might actually fix my current griddle/locker/etc predicament!
- [ ] Make the search-and-[kill/delete] code in ants & cockroaches into its own component/element/whatever an AI datum is.

**Comment by Wallemations** [comment]
Oh great there's a conflict and it doesn't tell me any details as to why it's got one, I had no errors compiling.
I'm guessing the Map Diff check is for maps, which is weird, because I didn't edit any maps.
Just in case though, I'll open up a couple maps and see if there's a conflict really quick, maybe an item changed paths & mine has the old pathing or something.

**Comment by Wallemations** [comment]
Yep checked all the station maps, no errors, no clue why that check failed.
If anyone knows please tell me

**Comment by Wallemations** [comment]
Okay, the issue was a signal I'm not even using anymore.
My main issues now are trying to work with `connect_loc` as well as the fact that the ant decal isn't damaging anyone anymore.
Fun!

**Comment by tf-4** [comment]
`connect_loc` is fucky atm

**Comment by Wallemations** [comment]
ah right, I forgot I removed the signal from the locker code, but I didn't merge that here, so it's still fucking it up, just give me a second.

**Comment by Wallemations** [comment]
> `connect_loc` is fucky atm

If that's the reason this is fuckin' up I guess there's nothing I can do at the moment.

**Comment by tf-4** [comment]
> 
> 
> > `connect_loc` is fucky atm
> 
> If that's the reason this is fuckin' up I guess there's nothing I can do at the moment.

i dunno if it's the reason for this, I haven't looked at the tests. But there's this https://github.com/tgstation/tgstation/issues/59426 and I've seen it cause test failures for PRs for a little while

**Comment by Wallemations** [comment]
This commit doesn't function and currently just softlocks the game without any runtime errors.
I need to swap computers which is why I even committed it.

**Comment by tralezab** [comment]
You should make a new pr instead of reopening this ancient one

**Comment by Wallemations** [comment]
> You should make a new pr instead of reopening this ancient one

I'm allowed to do that?
Aight, I'll do that when I get to the other computer then.

**Comment by Wallemations** [comment]
Going to make a new PR tomorrow since I was told to do so, I hope to god that's allowed. Hopefully it gets a little more pep in my step to fix whatever the hell could be causing these merge errors, and to figure out that AI code.

**Comment by ArcaneMusic** [comment]
Oh yeah, people make new PRs for older content all the time. You're fine.

**Pull request closed without merging**